applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_geo_steady_state_piping_element.cpp

```diff
@@ -0,0 +1,293 @@
+
+    PointerVector<Node> nodes;
+    nodes.push_back(r_model_part.CreateNewNode(1, 0.0, 0.0, 0.0));
+    nodes.push_back(r_model_part.CreateNewNode(2, 1.0, 0.0, 0.0));
+    const auto p_geometry = std::make_shared<Line2D2<Node>>(nodes);
+    return GeoSteadyStatePwPipingElement<2, 2>{1, p_geometry, rProperties};
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+using namespace Kratos;
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementIsAnElement, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const GeoSteadyStatePwPipingElement<2, 2> element;
+    auto p_casted_element = dynamic_cast<const Element*>(&element);
+    KRATOS_CHECK_NOT_EQUAL(p_casted_element, nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementCanCreateInstanceWithGeometryInput,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const GeoSteadyStatePwPipingElement<2, 2> element;
+    const auto p_geometry   = std::make_shared<Line2D2<Node>>(CreateNodes());
+    const auto p_properties = std::make_shared<Properties>();
+
+    // Act
+    const auto p_created_element = element.Create(1, p_geometry, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementCanCreateInstanceWithNodeInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto nodes        = CreateNodes();
+    const auto p_properties = std::make_shared<Properties>();
+
+    // The source element needs to have a geometry, otherwise the version of the
+    // Create method with a node input will fail.
+    const auto p_geometry = std::make_shared<Line2D2<Node>>(CreateNodes());
+    const GeoSteadyStatePwPipingElement<2, 2> element(0, p_geometry, p_properties);
+
+    // Act
+    const auto p_created_element = element.Create(1, nodes, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoSteadyStatePwPipingElementReturnsTheExpectedDoFList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_properties = std::make_shared<Properties>();
+
+    Model model;
+    const auto element = CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithPWDofs(model, p_properties);
+
+    // Act
+    const auto              dummy_process_info = ProcessInfo{};
+    Element::DofsVectorType degrees_of_freedom;
+    element.GetDofList(degrees_of_freedom, dummy_process_info);
+
+    // Assert
+    KRATOS_EXPECT_EQ(degrees_of_freedom.size(), 2);
+    for (auto dof : degrees_of_freedom) {
```

Comment:
I think `dof` is actually a pointer, so I reckon we should rename this variable to `p_dof`.
```suggestion
    for (auto p_dof : degrees_of_freedom) {
```

Reply:
Indeed DofsVectorType is a vector of pointers, so an element is a pointer
Done